src/components/routerLink.vue

```diff
@@ -4,14 +4,17 @@
   </a>
 </template>
 
-<script setup lang="ts" generic="T extends keyof RegisteredRouteMap">
+<script setup lang="ts" generic="T extends keyof RegisteredRouteMap | Url">
   import { computed, readonly } from 'vue'
   import { useRouter } from '@/compositions'
-  import { RegisteredRouteMap, RegisteredRouteWithParams } from '@/types/register'
+  import { RegisteredRouteMap, RegisteredRoutes } from '@/types/register'
   import { RouterPushOptions } from '@/types/routerPush'
+  import { RouteParamsByName } from '@/types/routeWithParams'
+  import { Url, isUrl } from '@/types/url'
 
   const props = defineProps<{
-    to: string | RegisteredRouteWithParams<T>,
+    to: T,
+    params?: T extends keyof RegisteredRouteMap ? RouteParamsByName<RegisteredRoutes, T> : undefined,
```

Comment:
params are always optional? That's probably not good. But we're going to struggle here because of vue's type support for props...

Reply:
ah yeah, feels like a lose-lose situation. you'll get a param invalid error if you don't pass in the params required, maybe we can do that check inside of router-link and provide a more helpful error?